### PR TITLE
AMQP source

### PR DESCRIPTION
**Pull request by mthenw**

### CODE REVIEW

File: workerpool/pool.go [hunk 1]
```diff
@@ -140,7 +140,7 @@ func newJob(session *concurrency.Session, config *connection.Job, locksPrefix st
 func (j *job) start() {
 	for i := uint(0); i < j.numWorkers; i++ {
 		id := uint(j.id.JobNumber()*j.bucketSize) + i
-		worker := newWorker(id, j.connection, j.waitGroup, j.log.Named("worker"))
+		worker := newWorker(id, *j.connection, j.waitGroup, j.log.Named("worker"))
```
Comment (sebito91) [inline workerpool/pool.go:143]: Is this why you were asking earlier about each worker getting a copy of the Connection? I'm not sure I like the pointer deref here, would rather keep the pointer intact and just use that. In this case they're not consuming anything that could potentially clobber the other workers.

**Review by sebito91 — COMMENTED** [review]
You just really want to get rid of that kafka source, eh? :D

File: workerpool/pool.go (hunk 1 above)
Comment (mthenw) [reply workerpool/pool.go:143]: > would rather keep the pointer intact and just use that

What do you mean?

**Comment by mthenw** [comment]
> You just really want to get rid of that kafka source, eh? :D

there is nothing there :)

**Review by sebito91 — APPROVED** [review]

**Pull request merged**